File: C++/11_재귀/Problem_03.cpp
/* 11단계 [ 재귀 ] */
// 제목 : 재귀의 귀재
// 문제 번호 : 25501

#include <iostream>
using namespace std;

int main(void)
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int N;
    cin >> N;

    for (int i = 0; i < N; i++)
    {
        string s;
        cin >> s;

        int cnt = 0, left = 0, right = s.length() - 1;
        bool flag = true;

        while (flag && left <= right)
        {
            cnt++;
            if (s[left] != s[right])
                flag = false;
            left++, right--;
        }
        cout << (flag ? 1 : 0) << " " << (flag && s.length() % 2 == 0 ? cnt + 1 : cnt) << "\n";
    }
    return 0;
}

File: Python/11_재귀/Problem_03.py
for i in range(int(input())):
    s = input()
    cnt, left, right, flag = 0, 0, len(s)-1, True

    while flag and left <= right:
        cnt += 1
        if s[left] != s[right]:
            flag = False
        left, right = left+1, right-1

    if flag:
        print(1, end=' ')
    else:
        print(0, end=' ')

    if not flag or len(s) % 2:
        print(cnt)
    else:
        print(cnt+1)
